add workspace load

enricosada opened the pull request:
ref https://github.com/fsharp/FsAutoComplete/issues/192

add a new command `workspaceload` to load:

- ~sln~
- [x] list of projects

Notes:

- [x] work is done in background
- ~use `dotnet.proj` parser for verbose sdk (old fsproj) instead of projectcracker~
- [x] notification to client for progress when a project is parsed
- [x] add some speed improvement (ref https://github.com/fsharp/FsAutoComplete/issues/161 ) to both dotnet/sdk and verbose sdk.
- [x] http project notification in another endpoint (request by @Krzysztof-Cieslak )

enricosada committed:
websocket on `/notify` with mailbox

use a mailboxprocess to queue the send of notification to websocket
send hello notification

Krzysztof-Cieslak commented:
> notification to client for progress when a project is parsed

You can implement that based on the notifications implementation I've done for the background project type checking. I think we should use separate endpoints for different types of push messages - some clients may want to listen for the type checking notifications but don't have as huge project managing capabilities as Ionide so may don't care about project parsing.

enricosada committed:
add workspaceLoad async command

load all projects passed, async
the progress (in progress/loaded/failed) is sent as notification

for stdio mode, disable background errors notifications

use System.Net.WebSockets.Client to test http notification

multiple notification sockets in /notify and /notifyWorkspace

enricosada committed:
stdio notifications with mailbox

enricosada commented:
ok done, merging

- http mode: this doesnt change current behaviour, and add the `/notifyWorkspace` for the new `workspaceLoad` notifications
- stdio mode: workspaceload notification are usual output, so is like a normal command

there are test for for both stdio (well, notificaiton here is normal output) and http mode (using a websocket client)
